nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -17,987 +17,569 @@
-    public String getTN(int iDim) {
-        String tn;
-        if (Tn[iDim] != null) {
-            tn = Tn[iDim];
+        if (dim == 0) {
+            int aqMod = block.optional($AQ_MOD).map(JCampRecord::getInt).orElse(0);
+            if (aqMod == 2)
+                return "rft";
         } else {
-            tn = getPar(".OBSERVENUCLEUS," + (iDim + 1));
-            if (tn == null) {
-                double sf = getSF(iDim);
-                tn = NMRDataUtil.guessNucleusFromFreq(sf).get(0).toString();
-            } else if ((tn.length() > 1) && (tn.charAt(0) == '^')) {
-                tn = tn.substring(1);
-            }
-            if (tn == null) {
-                tn = "";
+            FnMode fnMode = getFnMode(dim);
+            if (fnMode == FnMode.QSEQ || fnMode == FnMode.TPPI) {
+                return "rft";
+            } else if (fnMode == FnMode.UNDEFINED || fnMode == FnMode.STATES_TPPI) {
+                return "negate";
             }
-            Tn[iDim] = tn;
         }
-        return tn;
-    }
 
-    @Override
-    public String getSolvent() {
-        String s;
-        if ((s = getPar("SOLVENTNAME,1")) == null) {
-            s = "";
-        }
-        return s;
+        return "ft";
     }
 
     @Override
-    public double getTempK() {
-        Double d;
-        if ((d = getParDouble("TEMPERATURE,1")) == null) {
-// fixme what should we return if not present, is it ever not present
-            d = 298.0;
+    public double[] getCoefs(int dim) {
+        //XXX Was not implemented in original JCAMPData.
+        //Inspired from BrukerData instead.
+        if (dim == 0) {
+            return new double[0];
         }
-        return d;
-    }
 
-    @Override
-    public String getSequence() {
-        String s;
-        if ((s = getPar("PULPROG,1")) == null) {
-            s = "";
+        FnMode fnMode = getFnMode(dim);
+        if (fnMode == null || fnMode == FnMode.QF || fnMode == FnMode.QSEQ || fnMode == FnMode.TPPI) {
+            return new double[0];
+        } else if (fnMode == FnMode.STATES) {
+            return new double[]{1, 0, 0, 0, 0, 0, 1, 0};
+        } else if (fnMode == FnMode.UNDEFINED || fnMode == FnMode.STATES_TPPI) {
+            return new double[]{1, 0, 0, 0, 0, 0, 1, 0};
+        } else if (fnMode == FnMode.ECHO_ANTIECHO) {
+            return new double[]{1, 0, -1, 0, 0, 1, 0, 1};
         }
-        return s;
+        return new double[]{1, 0, 0, 1};
     }
 
     @Override
-    public long getDate() {
-        String s;
-        long seconds = 0;
-        if ((s = getPar("DATE,1")) != null) {
-            try {
-                System.out.println("sec " + s);
-                seconds = Long.parseLong(s);
-            } catch (NumberFormatException e) {
-            }
-        } else {
-            System.out.println("no date");
+    public String getSymbolicCoefs(int dim) {
+        //XXX Was not implemented in original JCAMPData.
+        //Inspired from BrukerData instead.
```

Comment:
Same: Not completely sure what this is supposed to do, especially because it was not implemented in the original JCAMPData.

Reply:
Different vendors collect data in different orders (which is especially important for 3D and higher).  If d refers to t1, t2, etc delays, and p refers to phases (real imaginary) and the number refers to the dimension.
Bruker might collect p1,d1,p2,d2
Varian might collect p1,p2,d1,d2
where the first item is incremeted fastest.
This symbolic pattern is used to setup the MultiVecCounter which determines the position vectors are read from dataset.